enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/hooks/use-get-current-resource.ts

```diff
@@ -0,0 +1,83 @@
+import { useAsync } from "react-use";
+
+import { skipToken, useGetCardQuery } from "metabase/api";
+import { fetchDashboard } from "metabase/dashboard/actions";
+import { getDashboardComplete } from "metabase/dashboard/selectors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import type {
+  SdkIframeEmbedSetupExperience,
+  SdkIframeEmbedSetupSettings,
+} from "metabase-enterprise/embedding_iframe_sdk_setup/types";
+import type { Card, Dashboard } from "metabase-types/api";
+
+const getResource = ({
+  experience,
+  dashboard,
+  card,
+}: {
+  experience: SdkIframeEmbedSetupExperience;
+  dashboard: Dashboard | null | undefined;
+  card: Card | null | undefined;
+}) => {
+  const isResourceWithDifferentTypeLoading =
+    (dashboard && experience !== "dashboard") ||
+    (card && experience !== "chart");
+
+  if (isResourceWithDifferentTypeLoading) {
+    return null;
+  }
+
+  return dashboard ?? card ?? null;
+};
+
+export const useGetCurrentResource = ({
+  experience,
+  settings,
+}: {
+  experience: SdkIframeEmbedSetupExperience;
+  settings: SdkIframeEmbedSetupSettings;
+}): {
+  resource: Dashboard | Card | null;
+  isLoading: boolean;
+  isFetching: boolean;
+} => {
+  const dispatch = useDispatch();
+
+  const { loading: isDashboardLoading } = useAsync(async () => {
+    if (!settings.dashboardId) {
+      return;
+    }
+
+    await dispatch(
+      fetchDashboard({
+        dashId: settings.dashboardId as number,
+        queryParams: {},
+      }),
+    );
+  }, [settings.dashboardId, dispatch]);
+  const dashboard = useSelector(getDashboardComplete);
```

Comment:
See https://metaboat.slack.com/archives/C505ZNNH4/p1759486836074399 about why i did it like this and not using RTK query